src/plugins/intel_gpu/src/graph/include/custom_gpu_primitive_inst.h

```diff
@@ -19,12 +20,37 @@ class typed_primitive_inst<custom_gpu_primitive> : public typed_primitive_inst_b
 
 public:
     template<typename ShapeType>
-    static std::vector<layout> calc_output_layouts(custom_gpu_primitive_node const& /*node*/, const kernel_impl_params& impl_param) {
+    static std::vector<layout> calc_output_layouts(custom_gpu_primitive_node const& node, const kernel_impl_params& impl_param) {
         assert(static_cast<bool>(impl_param.desc->output_data_types[0]) == false &&
                "Output data type forcing is not supported for "
                "custom_gpu_primitive_node!");
         layout output_layout = impl_param.typed_desc<custom_gpu_primitive>()->output_layout;
 
+        bool is_dynamic = false;
+        const auto inp_sz = impl_param.get_input_layout_size();
+        for (size_t i = 0; i < inp_sz; i++) {
+            if (impl_param.get_input_layout(i).is_dynamic()) {
+                is_dynamic = true;
+                break;
+            }
+        }
+
+        if (!is_dynamic && output_layout.is_dynamic()) {
+            ov::OutputVector new_inputs;
+            for (size_t i = 0; i < inp_sz; i++) {
+                auto dt = impl_param.get_input_layout(i).data_type;
+                // std::make_shared<ov::op::v0::Parameter>(inType, inputDynamicShapes[0])
+                auto input = std::make_shared<ov::op::v0::Parameter>(dt, impl_param.get_input_layout(i).get_shape());
+                new_inputs.emplace_back(input);
+            }
+
+            auto op = impl_param.typed_desc<custom_gpu_primitive>()->op;
+            auto new_op = op->clone_with_new_inputs(new_inputs);
+            new_op->validate_and_infer_types();
```

Comment:
In rare case, the shape infer may depend on the input data and it should not work here. Maybe we can detect the case and report error to the user?

Reply:
hi @luo-cheng2021 , I found [our doc](https://github.com/openvinotoolkit/openvino/blob/c168ebba59ec796e69f0b2b927cdb28bb8d9960c/docs/articles_en/documentation/openvino-extensibility/custom-openvino-operations.rst?plain=1#L34) just highlight use ``get_input_partial_shape()`` and ``get_input_element_type()`` to inference output shape. I think it maybe enough.